### PR TITLE
Fix (Core) Resolve issue with dialog options randomly not working

### LINKED ISSUE #15989: Gossips on NPCs randomly turn unusable at random
### Current Behaviour

All NPCs except those who go directly into Vendor/Training window can reproduce this bug.

The problem is when interacting with a NPC that has a gossip in certain random situations the gossip is unusable. Options are shown, but when clicked they do nothing. After closing the window, and interacting again it gets fixed.
Following up on discord, seems i'm not the only one seeing this behaviour: https://discord.com/channels/217589275766685707/284406375495368704/1097561125052223649

### Expected Blizzlike Behaviour

Gossips should work 10/10 you try

### Source

_No response_

### Steps to reproduce the problem

This is not so easy to reproduce since it happens at random, seems to be related to approching NPCs to fast or circleing through the menu options fast.

### Extra Notes

_No response_

### AC rev. hash/commit

AzerothCore rev. fd6941de7c7e 2023-04-09 23:44:52 -0300 (prod branch) (Unix, RelWithDebInfo, Static)
Using SSL version: OpenSSL 1.1.1n  15 Mar 2022 (library: OpenSSL 1.1.1n  15 Mar 2022)
Using Boost version: 1.74.0
Using MySQL version: 100518
Using CMake version: 3.18.4
Compiled on: Linux 5.10.0-20-amd64
Worldserver listening connections on port 8085
Realmlist (Realm Id: 1) configured in port 8085
VMAPs status: Enabled. LineOfSight: 1, getHeight: 1, indoorCheck: 1
MMAPs status: Enabled
maps directory located in /root/azeroth-server/data/maps. Total size: 291014951 bytes
vmaps directory located in /root/azeroth-server/data/vmaps. Total size: 658130721 bytes
mmaps directory located in /root/azeroth-server/data/mmaps. Total size: 2192910800 bytes
Using esES DBC Locale as default. All available DBC locales: esES 
Using World DB: ACDB 335.9-dev
LoginDatabase queue size: 0
CharacterDatabase queue size: 0
WorldDatabase queue size: 0
> List enable modules:
- mod-eluna
- mod-globalchat
- mod-ip-tracker
- mod-multi-client-check
- mod-npc-spectator
- mod-reward-shop
- mod-transmog
- mod-weekend-xp

### Operating system

Debian 11

### Custom changes or Modules

- mod-eluna
- mod-globalchat
- mod-ip-tracker
- mod-multi-client-check
- mod-npc-spectator
- mod-reward-shop
- mod-transmog
- mod-weekend-xp

### PR DESCRIPTION
**Pull request by kjack9**
## Changes Proposed:

This PR proposes changes to:
-  [X] Core (units, players, creatures, game systems).
-  [ ] Scripts (bosses, spell scripts, creature scripts).
-  [ ] Database (SAI, creatures, etc).

https://github.com/azerothcore/azerothcore-wotlk/issues/15989#issuecomment-1782229624

## Issues Addressed:

- Closes #15989
- Closes https://github.com/chromiecraft/chromiecraft/issues/3997

## SOURCE:

The changes have been validated through:
- [ ] Live research (checked on live servers, e.g Classic WotLK, Retail, etc.)
- [ ] Sniffs (remember to share them with the open source community!)
- [ ] Video evidence, knowledge databases or other public sources (e.g forums, Wowhead, etc.)
- [ ] The changes promoted by this pull request come partially or entirely from another project (cherry-pick). **Cherry-picks must be committed using the proper --author tag in order to be accepted, thus crediting the original authors, unless otherwise unable to be found**

## Tests Performed:

This PR has been:
- [X] Tested in-game by the author.
- [X] Tested in-game by other community members/someone else other than the author/has been live on production servers.
- [ ] This pull request requires further testing and may have edge cases to be tested.

## How to Test the Changes:

- [ ] This pull request can be tested by following the reproduction steps provided in the linked issue
- [ ] This pull request requires further testing. Provide steps to test your changes. If it requires any specific setup e.g multiple players please specify it as well.

1. `.go creature 116913`
2. Moving left to right breaks the dialog before this change

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: src/server/game/Scripting/ScriptDefines/CreatureScript.cpp [hunk 1]
```diff
@@ -148,7 +148,6 @@ uint32 ScriptMgr::GetDialogStatus(Player* player, Creature* creature)
     ASSERT(creature);
 
     auto tempScript = ScriptRegistry<CreatureScript>::GetScriptById(creature->GetScriptId());
-    ClearGossipMenuFor(player);
```
Comment (pklloveyou) [inline src/server/game/Scripting/ScriptDefines/CreatureScript.cpp:151]: GossipMenu& gossipMenu = player->PlayerTalkClass->GetGossipMenu();
    if (gossipMenu.GetSenderGUID() == creature->GetGUID())
        ClearGossipMenuFor(player);

Comment (pklloveyou) [reply src/server/game/Scripting/ScriptDefines/CreatureScript.cpp:151]: What do you think of doing this？

Comment (kjack9) [reply src/server/game/Scripting/ScriptDefines/CreatureScript.cpp:151]: @AnchyDev @heyitsbench @lewpar

I was told this PR would be free? =D

Comment (AnchyDev) [reply src/server/game/Scripting/ScriptDefines/CreatureScript.cpp:151]: That's probably okay, if you wanna test it. You just don't want to clear if the GUID is not the same.

**Comment by kjack9** [comment]
As an updated, I've been using this on my server for the last 5 days and haven't had a missed dialog, nor a broken one.

**Comment by AnchyDev** [comment]
> As an update, I've been using this on my server for the last 5 days and haven't had a missed dialog, nor a broken one.

I've got it patched on my local server and have not noticed it happen while using my gossip NPCs also.

**Review by Nyeriah — APPROVED** [review]

**Comment by Nyeriah** [comment]
Merging as is if needed make a new PR for the changes suggested

**Pull request merged**